ArpyEnhance.user.js

```diff
@@ -468,6 +468,15 @@ ciggar
       }
     });
 
+	// TODO item
+	let todoItemElement = document.getElementById("todo_item_id");
+	let selectedTodoItemElementText = todoItemElement.options[todoItemElement.selectedIndex].text;
+	let ersteRedmineTicketIndex = selectedTodoItemElementText.indexOf("#");
+	let ersteRedmineTicket = null;
+	if ( ersteRedmineTicketIndex !== -1 ) {
+		ersteRedmineTicket = selectedTodoItemElementText.substring(ersteRedmineTicketIndex + 1, ersteRedmineTicketIndex + 6);
+	}
+
```

Comment:
De a nagyobb gond ezzel a fajta megközelítéssel, hogy csak és kizárólag a lenyíló mezőkben kiválasztott értéket veszed figyelembe.

A feldolgozás működési elve az az, hogy alapból a lenyílókban kiválasztott "project_id / todo_list / todo_item" hármasnak csak a legelső (előzőleg kedvencként elmentett) kategória címke előfordulásáig van hatása.
Viszont a te megoldásod az összes munkaidő bejegyzés sorra érvényesül, felülírva (vagy inkább beelőzve) az adott sorra érvényes címkét, és a sorban helyileg a leírásban detektálható ticket számot.

Habár ahogy fentebb írtam, a lenyílókból kiválasztott értékek érvényesülnek a legelsőnek megadott, címkétlen sorokra, a gyakorlatban szerintem senki sem használja így az Arpy Enhance-t.
A script lényege, hogy minél kevesebb mikromenedzsmenttel lehessen megoldani a munkaórák felvitelét. Így a leggyakrabban a lenyílókat csupán arra használjuk, hogy elmentsük az értékeket címkékbe.

Tehát azt is jó lenne támogatni, hogy címkeként elmentett todo-ból is elő tudjon jönni a ticket szám.
Aztán jó lenne ezt minél generikusabban is megoldani, így az elnevezésekben nem használnám az erste szót. Igazából bármelyik projektnél hasznos lehet ez a feature. (-> Youtrack ticket formátum támogatása, ami pl. "MHT-1306", tehát a formátum: valamennyi csupa nagy betű, kötőjel, aztán valamennyi szám)
Továbbmenve, azt se feltételezzük, hogy csak a todo item-ben lehet ticket szám. Miért ne fordulhatna elő a másik kettő kiválasztható értéknél is?
A prioritásnak pedig úgy kellene lennie, hogy a specifikusabb ticket szám nyer, tehát minél lentebbi szinten van, annál erősebb: project_id < todo_list < todo_item < sor